api/src/main/java/org/apache/iceberg/metrics/DefaultMetricsContext.java

```diff
@@ -22,6 +22,7 @@
 
 /** A default {@link MetricsContext} implementation that uses native Java counters/timers. */
 public class DefaultMetricsContext implements MetricsContext {
+  private static final int DEFAULT_HISTOGRAM_RESERVOIR_SIZE = 1024;
```

Comment:
How about using 10k, rather than 1k? If we use reservoir sampling, then that would give us about 100 data points for percentiles, which seems like a good enough default.

Reply:
I am fine with 10K as default value. 10k size long[] array would cost 80 KB per metric. 100 histogram metrics would be 8 MB, which isn't too bad.